templates/lb-fargate-service/cf.yml

```diff
@@ -1,168 +1,174 @@
 # Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
-
-# TODO: handle conditional Load Balancer in environment
-# TODO: Parameterize TaskDefinition/Service constructs
-
+AWSTemplateFormatVersion: 2010-09-09
+Description: CloudFormation template that represents an load balanced web application.
 Parameters:
-  EnvironmentName:
-    Description: The environment stack to deploy the application stack into.
+  ProjectName:
     Type: String
-
-  Image:
-    Description: The Docker image to run.
+    Default: {{.ProjectName}}
+  EnvName:
     Type: String
-
+    Default: {{.EnvName}}
+  AppName:
+    Type: String
+    Default: {{.AppName}}
+  ContainerImage:
+    Type: String
+    Default: {{.Image.Name}}
+  ContainerPort:
+    Type: Number
+    Default: {{.Image.Port}}
+  RulePath:
+    Type: String
+    Default: {{.Path}}
+  RulePriority:
+    Type: Number
+    Default: {{.PathPriority}} # This number must be unique between rules.
+  TaskCPU:
+    Type: String
+    Default: {{.CPU}}
+  TaskMemory:
+    Type: String
+    Default: {{.Memory}}
   TaskCount:
     Type: Number
-    Default: 2
-
+    Default: {{.Count}}
 Resources:
-  SecurityGroupEgress:
-    Type: AWS::EC2::SecurityGroupEgress
-    Properties:
-      GroupId:
-        Fn::ImportValue: !Sub ${EnvironmentName}-PublicLoadBalancerSecurityGroupId
-      IpProtocol: tcp
-      Description: Load balancer to target
-      DestinationSecurityGroupId: !GetAtt SecurityGroup.GroupId
-      FromPort: 80
-      ToPort: 80
-
-  Listener:
-    Type: AWS::ElasticLoadBalancingV2::Listener
-    Properties:
-      DefaultActions:
-        - TargetGroupArn: !Ref TargetGroup
-          Type: forward
-      LoadBalancerArn:
-        Fn::ImportValue: !Sub ${EnvironmentName}-PublicLoadBalancerArn
-      Port: 80
-      Protocol: HTTP
-
-  TargetGroup:
-    Type: AWS::ElasticLoadBalancingV2::TargetGroup
-    Properties:
-      Port: 80
-      Protocol: HTTP
-      TargetType: ip
-      VpcId:
-        Fn::ImportValue: !Sub ${EnvironmentName}-VpcId
-
-  TaskRole:
-    Type: AWS::IAM::Role
+  LogGroup: # NOTE Should we prefix the resource logical names with environment name + app name?
```

Comment:
yes plz

Reply:
Done, the only gotcha is that now we have to add a `clean` function to the template ([example](https://forum.golangbridge.org/t/how-to-call-a-custom-function-in-golang-template/4934)) because it's possible that the `ProjectName`, `EnvName`, or `AppName` has the dash character `-` which is not allowed in logical IDs: (a-z, A-Z, 0-9).

The `clean` function will just remove those characters from the string.

Reply:
What does having the `ProjectName, `EnvName, and `AppName` values as part of the logical ID do for us exactly?

Reply:
Good question! I don't think it does much for us in this scenario actually 🤔

Usually we want these logical ids to be pretty specific, since you can't have have two resources with the same logical id. But I don't think we expect more than one of these resource so it might not be necessary.

I'll keep it as is for now and we can change it afterwards

Reply:
I would advocate for removing it if it provides no clear value currently as I find it difficult to read, but will leave it up to you.